client/app/services/query-result.js

```diff
@@ -147,7 +150,12 @@ function QueryResultService($resource, $timeout, $q) {
       Job.delete({ id: this.job.id });
     }
 
-    getStatus() {
+    getStatus(extendedStatus) {
+      if (extendedStatus) {
```

Comment:
It was the simplest way to retain backward compatibility. There is some other code that uses `getStatus()` and relies on pre-defined "server" statuses, and I don't want to accidentally break it. So I added this special case - if somewhere we need to know more detailed status of query - we should pass parameter and `getStatus()` will return it; existing code will keep working in old way. Another idea was to add method - something like `isLoadingData()`, but I think this version is better

Reply:
There are 9 uses in 6 files. All of them seem to check for specific cases and will work with this case.

While it will require another test, I think it's worth it as we gain simpler code to maintain.

Reply:
Okay, I'll check all uses of `getStatus`, and, if possible, simplify this function. Will ping you when done.